gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/GradleJdkInstallationSetup.java

```diff
@@ -121,16 +123,39 @@ private static boolean copy(ILogger logger, Path destinationJdkInstallationDirec
                 lockFile, StandardOpenOption.READ, StandardOpenOption.CREATE, StandardOpenOption.WRITE)) {
             channel.lock();
             // double-check, now that we hold the lock
-            if (Files.exists(destinationJdkInstallationDirectory)) {
+            if (Files.exists(destinationJdkInstallationDirectory)
+                    && Files.exists(destinationJdkInstallationDirectory.resolve("bin/java"))) {
                 logger.log(String.format("Distribution URL %s already exists", destinationJdkInstallationDirectory));
                 return false;
             }
             logger.log(
                     String.format("Copying JDK from %s into %s", currentJavaHome, destinationJdkInstallationDirectory));
-            FileUtils.copyDirectory(currentJavaHome, destinationJdkInstallationDirectory);
+            Path tempDirWithPrefix = Files.createTempDirectory("gradle-jdks");
```

Comment:
Atomic moves generally only work when the source and destination for the filesystem object are on the same filesystem. If they're not it's usually actually a copy. I think here this risks copying to a temporary filesystem, then copying again rather than moving to the final destination.

To work around this, I'd make a `tmp` dir at `~/.gradle/gradle-jdks/tmp` and copy it to a subdir in there first before moving.